### PR TITLE
Urgent fixes

### PR DESCRIPTION
**Pull request by itsmuntadhar**
This PR fixes windows support not working properly after #62 since the directive `WINDOWS` is not being used

Also added a way to list printers

### CODE REVIEW

File: PDFtoPrinter/GetPrinters.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Management;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace PDFtoPrinter
+{
+    public sealed class GetPrinters
```
Comment (svishnevsky) [inline PDFtoPrinter/GetPrinters.cs]: Hi @itsmuntadhar ,
Thanks for your contribution! It's just a small one. All the other classes in the project are nouns, and this one is more like a verb. Can you rename the class to something like "PrintersQuery" and the method name below "Run" or a similar concept just to keep the source code consistent?

**Review by svishnevsky — CHANGES_REQUESTED** [review]

File: PDFtoPrinter/GetPrinters.cs (hunk 1 above)
Comment (itsmuntadhar) [reply PDFtoPrinter/GetPrinters.cs]: sure, will do it ASAP

Comment (itsmuntadhar) [reply PDFtoPrinter/GetPrinters.cs]: @svishnevsky update is live now.
hope you find time to review it again.

**Review by svishnevsky — APPROVED** [review]

**Comment by svishnevsky** [comment]
Hi @itsmuntadhar 
Unfortunately the build failed:

```
 C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\Roslyn\csc.exe /noconfig /unsafe- /checked- /nowarn:1701,1702,1701,1702 /fullpaths /nostdlib+ /errorreport:prompt /warn:4 /define:TRACE;RELEASE;NETFRAMEWORK;NET46;NET20_OR_GREATER;NET30_OR_GREATER;NET35_OR_GREATER;NET40_OR_GREATER;NET45_OR_GREATER;NET451_OR_GREATER;NET452_OR_GREATER;NET46_OR_GREATER /highentropyva+ /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6\mscorlib.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6\System.Core.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6\System.Data.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6\System.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6\System.Drawing.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6\System.IO.Compression.FileSystem.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6\System.Numerics.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6\System.Printing.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6\System.Runtime.Serialization.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6\System.Xml.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6\System.Xml.Linq.dll" /debug- /debug:portable /filealign:512 /optimize+ /out:"obj\any cpu\release\net46\PDFtoPrinter.dll" /subsystemversion:6.00 /resource:"obj\any cpu\release\net46\PDFtoPrinter.ErrorMessages.resources" /target:library /warnaserror- /utf8output /deterministic+ /sourcelink:"obj\any cpu\release\net46\PDFtoPrinter.sourcelink.json" /langversion:7.3 /analyzerconfig:D:\a\1\s\.editorconfig /analyzerconfig:"obj\any cpu\release\net46\PDFtoPrinter.GeneratedMSBuildEditorConfig.editorconfig" CleanupFilesPrinter.cs ErrorMessages.Designer.cs IPrinter.cs IPrinterStreamExtensions.cs IProcess.cs IProcessFactory.cs PDFtoPrinterPrinter.cs PDFtoPrintWrapper.cs PrintersQuery.cs PrintingOptions.cs Process.cs QueuedFile.cs StreamPrintingOptions.cs StringExtensions.cs SystemProcessFactory.cs "obj\any cpu\release\net46\.NETFramework,Version=v4.6.AssemblyAttributes.cs" "obj\any cpu\release\net46\PDFtoPrinter.AssemblyInfo.cs" /warnaserror+:NU1605
##[error]PDFtoPrinter\SystemProcessFactory.cs(13,17): Error CS0103: The name 'RuntimeInformation' does not exist in the current context
D:\a\1\s\PDFtoPrinter\SystemProcessFactory.cs(13,17): error CS0103: The name 'RuntimeInformation' does not exist in the current context [D:\a\1\s\PDFtoPrinter\PDFtoPrinter.csproj::TargetFramework=net46]
##[error]PDFtoPrinter\SystemProcessFactory.cs(13,49): Error CS0103: The name 'OSPlatform' does not exist in the current context
D:\a\1\s\PDFtoPrinter\SystemProcessFactory.cs(13,49): error CS0103: The name 'OSPlatform' does not exist in the current context [D:\a\1\s\PDFtoPrinter\PDFtoPrinter.csproj::TargetFramework=net46]
##[error]PDFtoPrinter\PDFtoPrinterPrinter.cs(16,51): Error CS0103: The name 'RuntimeInformation' does not exist in the current context
D:\a\1\s\PDFtoPrinter\PDFtoPrinterPrinter.cs(16,51): error CS0103: The name 'RuntimeInformation' does not exist in the current context [D:\a\1\s\PDFtoPrinter\PDFtoPrinter.csproj::TargetFramework=net46]
##[error]PDFtoPrinter\PDFtoPrinterPrinter.cs(16,83): Error CS0103: The name 'OSPlatform' does not exist in the current context
D:\a\1\s\PDFtoPrinter\PDFtoPrinterPrinter.cs(16,83): error CS0103: The name 'OSPlatform' does not exist in the current context [D:\a\1\s\PDFtoPrinter\PDFtoPrinter.csproj::TargetFramework=net46]
##[error]PDFtoPrinter\PrintersQuery.cs(15,17): Error CS0103: The name 'RuntimeInformation' does not exist in the current context
D:\a\1\s\PDFtoPrinter\PrintersQuery.cs(15,17): error CS0103: The name 'RuntimeInformation' does not exist in the current context [D:\a\1\s\PDFtoPrinter\PDFtoPrinter.csproj::TargetFramework=net46]
##[error]PDFtoPrinter\PrintersQuery.cs(15,49): Error CS0103: The name 'OSPlatform' does not exist in the current context
D:\a\1\s\PDFtoPrinter\PrintersQuery.cs(15,49): error CS0103: The name 'OSPlatform' does not exist in the current context [D:\a\1\s\PDFtoPrinter\PDFtoPrinter.csproj::TargetFramework=net46]
##[error]PDFtoPrinter\PrintersQuery.cs(17,31): Error CS0246: The type or namespace name 'ManagementObjectSearcher' could not be found (are you missing a using directive or an assembly reference?)
D:\a\1\s\PDFtoPrinter\PrintersQuery.cs(17,31): error CS0246: The type or namespace name 'ManagementObjectSearcher' could not be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\PDFtoPrinter\PDFtoPrinter.csproj::TargetFramework=net46]
##[error]PDFtoPrinter\PrintersQuery.cs(18,26): Error CS0246: The type or namespace name 'ManagementBaseObject' could not be found (are you missing a using directive or an assembly reference?)
D:\a\1\s\PDFtoPrinter\PrintersQuery.cs(18,26): error CS0246: The type or namespace name 'ManagementBaseObject' could not be found (are you missing a using directive or an assembly reference?) [D:\a\1\s\PDFtoPrinter\PDFtoPrinter.csproj::TargetFramework=net46]
##[error]PDFtoPrinter\PDFtoPrinterPrinter.cs(89,17): Error CS0103: The name 'RuntimeInformation' does not exist in the current context
D:\a\1\s\PDFtoPrinter\PDFtoPrinterPrinter.cs(89,17): error CS0103: The name 'RuntimeInformation' does not exist in the current context [D:\a\1\s\PDFtoPrinter\PDFtoPrinter.csproj::TargetFramework=net46]
##[error]PDFtoPrinter\PDFtoPrinterPrinter.cs(89,49): Error CS0103: The name 'OSPlatform' does not exist in the current context
D:\a\1\s\PDFtoPrinter\PDFtoPrinterPrinter.cs(89,49): error CS0103: The name 'OSPlatform' does not exist in the current context [D:\a\1\s\PDFtoPrinter\PDFtoPrinter.csproj::TargetFramework=net46]
  CompilerServer: server - server processed compilation - PDFtoPrinter (net46)
Done Building Project "D:\a\1\s\PDFtoPrinter\PDFtoPrinter.csproj" (Build target(s)) -- FAILED.

```

**Pull request merged**